galaxyproject/galaxy
# [21.09] Use new sentry-sdk and adapt usage pattern

Replaces our custom Sentry WSGI middleware with middleware shipped by
sentry-sdk. Adds sentry-sdk ASGI middleware. Our custom sentry
middleware also allowed logging slow requests to sentry. I have dropped
this for now, and if we still want this in the future, we should
probably implement this as a custom logging middleware instead, which
would automatically appear in Sentry as well.

We used to hardcode the log level for events to WARNING, and the Sentry
default is ERROR now, which I think is a good idea. I've made this
configurable using the sentry_event_level config variable.

It does not seem to be possible to connect multiple dsn's in the same
app, unless we start hacking
(https://github.com/getsentry/sentry-python/issues/198), so I dropped
the custom_dsn from the sentry error reporting plugin (which has been
broken anyway, so shouldn't be a big deal).

## How to test the changes?
(Select all options that apply)
Spin up a sentry on-premise instance (https://github.com/getsentry/onpremise worked well for me) and set up the integration. Make sure logs and tool error reports appear in Sentry.

## License
